test/WrappingIterator-test.js

```diff
@@ -0,0 +1,116 @@
+import {
+  AsyncIterator,
+  ArrayIterator,
+  WrappingIterator,
+} from '../dist/asynciterator.js';
+
+import { EventEmitter } from 'events';
+
+import { Readable } from 'readable-stream';
+
+describe('WrappingIterator', () => {
+  describe('The WrappingIterator function', () => {
+    describe('the result when called with `new`', () => {
+      let instance;
+      before(() => { instance = new WrappingIterator(); });
+
+      it('should be an WrappingIterator object', () => {
+        instance.should.be.an.instanceof(WrappingIterator);
+      });
+
+      it('should be an AsyncIterator object', () => {
+        instance.should.be.an.instanceof(AsyncIterator);
+      });
+
+      it('should be an EventEmitter object', () => {
+        instance.should.be.an.instanceof(EventEmitter);
+      });
+    });
+  });
+
+  describe('with a stream.Readable source', () => {
+    let iterator;
+    let source;
+    before(() => {
+      source = Readable.from([0, 1, 2, 3, 4]);
+      iterator = new WrappingIterator(source);
+    });
+    it('should emit all items', async () => {
+      expect(await iterator.toArray()).to.deep.equal([0, 1, 2, 3, 4]);
+    });
+  });
+
+  describe('with a promisified stream.Readable source', () => {
+    let iterator;
+    let source;
+    before(() => {
+      source = Promise.resolve(Readable.from([0, 1, 2, 3, 4]));
```

Comment:
For the sake of checking you're handling events correctly please have a few more `Promise` tests with different source types, e.g. range iterators, actual arrays - and more importantly sources that will buffer.

Reply:
Done!